typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,30 @@
+const tabContainer = document.querySelector('.tab-container') as HTMLDivElement;
+const tabs = document.querySelectorAll('.tab') as NodeListOf<HTMLDivElement>;
+const views = document.querySelectorAll('.view') as NodeListOf<HTMLDivElement>;
+
```

Comment:
Try not to use Type Assertion unless it's absolutely necessary since it's essentially disables TypeScript by us saying we know what it is. You have your checks on line 5, 6, and 7 which would cover your error of needing to assert types.